1.image_quality_control/scripts/3.generate_qc_report.r

```diff
@@ -7,9 +7,37 @@ suppressPackageStartupMessages({
     library(colorspace)
 })
 
+# Get the current working directory and find Git root
+find_git_root <- function() {
+    # Get current working directory
+    cwd <- getwd()
+
+    # Check if current directory has .git
+    if (dir.exists(file.path(cwd, ".git"))) {
+        return(cwd)
+    }
+
+    # If not, search parent directories
+    current_path <- cwd
+    while (dirname(current_path) != current_path) {  # While not at root
+        parent_path <- dirname(current_path)
+        if (dir.exists(file.path(parent_path, ".git"))) {
+            return(parent_path)
+        }
+        current_path <- parent_path
+    }
+
+    # If no Git root found, stop with error
+    stop("No Git root directory found.")
+}
```

Comment:
Consider using reticulate or similar to leverage a single function once / where available (if you need it after experimenting with settings).